src/test/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/AssemblyBasedCallerUtilsUnitTest.java

```diff
@@ -0,0 +1,56 @@
+package org.broadinstitute.hellbender.tools.walkers.haplotypecaller;
+
+import htsjdk.samtools.SAMFileHeader;
+import htsjdk.samtools.SAMLineParser;
+import htsjdk.samtools.SAMReadGroupRecord;
+import htsjdk.samtools.SAMRecord;
+
+import org.broadinstitute.hellbender.engine.AssemblyRegion;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+import org.broadinstitute.hellbender.utils.genotyper.SampleList;
+import org.broadinstitute.hellbender.utils.read.ArtificialReadUtils;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+import org.broadinstitute.hellbender.utils.read.SAMRecordToGATKReadAdapter;
+
+import java.util.*;
+
+import org.broadinstitute.hellbender.GATKBaseTest;
+import org.testng.Assert;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+public class AssemblyBasedCallerUtilsUnitTest extends GATKBaseTest {
+    private SAMFileHeader header;
+
+    @BeforeClass
+    public void init() {
+        header = ArtificialReadUtils.createArtificialSamHeader(1, 1, 100000000);
+        SAMReadGroupRecord rg = new SAMReadGroupRecord("tumor");
+        rg.setSample("tumor");
+        header.addReadGroup(rg);
+    }
+
+    @Test
+    public void testfinalizeRegion() {
```

Comment:
This test looks good, though in this case since there is only one test in the AssemblyBasedCallerUtilsUnitTests I wouldn't use the `@BeforeClass` to initialize files. Instead just put them into static fields or build the header etc... into this test.

Reply:
Also this is a good test